src/record/record.js

```diff
@@ -125,34 +113,15 @@ Record.prototype.set = function( pathOrData, data ) {
 		return this;
 	}
 
-	if( arguments.length === 2 && utils.deepEquals( this._getPath( pathOrData ).getValue(), data ) ) {
-		return this;
-	}
-	else if( arguments.length === 1 && utils.deepEquals( this._$data, pathOrData ) ) {
-		return this;
-	}
-
-	this._beginChange();
-	this.version++;
+	var path = arguments.length === 1 ? undefined : pathOrData;
+	data = path ? data : pathOrData;
 
-	if( arguments.length === 1 ) {
-		this._$data = ( typeof pathOrData == 'object' ) ? utils.deepCopy( pathOrData ) : pathOrData;
-		this._connection.sendMsg( C.TOPIC.RECORD, C.ACTIONS.UPDATE, [
-			this.name,
-			this.version,
-			this._$data
-		]);
-	} else {
-		this._getPath( pathOrData ).setValue( ( typeof data == 'object' ) ? utils.deepCopy( data ): data );
-		this._connection.sendMsg( C.TOPIC.RECORD, C.ACTIONS.PATCH, [
-			this.name,
-			this.version,
-			pathOrData,
-			messageBuilder.typed( data )
-		]);
+	if ( utils.deepEquals( data, jsonPath.get( this._$data, path ) ) ) {
+		return this;
```

Comment:
This check ends up as a duplicate if there is a listener on that path. Not sure how to resolve that...